src/backend/dx12/src/lib.rs

```diff
@@ -1081,13 +1098,34 @@ impl hal::Instance<Backend> for Instance {
                     max_bound_descriptor_sets: MAX_DESCRIPTOR_SETS as u16,
                     max_descriptor_set_uniform_buffers_dynamic: 8,
                     max_descriptor_set_storage_buffers_dynamic: 4,
-                    max_per_stage_descriptor_sampled_images: d3d12::D3D12_COMMONSHADER_INPUT_RESOURCE_REGISTER_COUNT as _,
-                    max_per_stage_descriptor_samplers: d3d12::D3D12_COMMONSHADER_SAMPLER_REGISTER_COUNT as _,
-                    max_per_stage_descriptor_storage_buffers: 4,
-                    max_per_stage_descriptor_storage_images: 4,
-                    max_per_stage_descriptor_uniform_buffers: d3d12::D3D12_COMMONSHADER_CONSTANT_BUFFER_API_SLOT_COUNT as _,
+                    max_descriptor_set_sampled_images: full_heap_count,
+                    max_descriptor_set_storage_buffers: uav_limit,
+                    max_descriptor_set_storage_images: uav_limit,
+                    max_descriptor_set_uniform_buffers: full_heap_count,
+                    max_descriptor_set_samplers: d3d12::D3D12_MAX_SHADER_VISIBLE_SAMPLER_HEAP_SIZE as _,
+                    max_per_stage_descriptor_sampled_images: match features.ResourceBindingTier {
+                        d3d12::D3D12_RESOURCE_BINDING_TIER_1 => 128,
+                        d3d12::D3D12_RESOURCE_BINDING_TIER_2
+                        | d3d12::D3D12_RESOURCE_BINDING_TIER_3 => full_heap_count,
+                        _ => unreachable!(),
+                    } as _,
+                    max_per_stage_descriptor_samplers: match features.ResourceBindingTier {
+                        d3d12::D3D12_RESOURCE_BINDING_TIER_1 => 16,
+                        d3d12::D3D12_RESOURCE_BINDING_TIER_2
+                        | d3d12::D3D12_RESOURCE_BINDING_TIER_3 => d3d12::D3D12_MAX_SHADER_VISIBLE_SAMPLER_HEAP_SIZE,
+                        _ => unreachable!(),
+                    } as _,
+                    max_per_stage_descriptor_storage_buffers: uav_limit,
```

Comment:
that seems rather high, i.e. it's higher than `D3D12_UAV_SLOT_COUNT`

Reply:
The table says that tier 3 gets the "full heap", so I would imagine this would be okay.